### ISSUE TITLE
We should update DIKUArcade to conform with C# 7.0 ?

### ISSUE THREAD

**Issue by alexpanter** [diku-dk/DIKUArcade#32]
The newest patch for DIKUArcade (https://github.com/diku-dk/DIKUArcade/commit/b526f206308989028c41dfef31f3b9022bd48d41) relies upon the `ref` feature from C# 7, as in:
```csharp
ref var Tuple = ref events[i];
```
to reference a struct. Maybe there is another way of doing this, but this feature is central for this patch, which again is central for solving exercise 9.

@BorisDuedder would it be safe to opt for C# 7 ? Or can we (temporarily) change this code to do the same but at the same time adhere to C# 6 specification?

**Comment by BorisDuedder** [comment]
I think it is safe to use the C# 7 feature here. @alexpanter Please add a comment to explain the decision.

**Comment by Ulrik-dk** [comment]
@BorisDuedder does this mean we will have acces to other C# 7 features in the SpaceTaxi projects?

**Comment by alexpanter** [comment]
@BorisDuedder I will make an announcement on Absalon - I suppose this is what you mean by 'comment'.

**Comment by Mystiking** [comment]
Hey Boris and Alexander.
Is this the correct forum for discussing this chance to DIKUArcade? Or can
we wait for the TA meeting friday?

**Comment by alexpanter** [comment]
The students need it for their next hand-in, so there really is no waiting. Did you have problems getting C# 7 to work on your machine?

**Comment by Mystiking** [comment]
Yes, and I know a lot of the students running Linux are as well. We have
also stated that code written using C# 7 features will not be corrected in
previous assignments...

**Comment by WilliamDue** [comment]
The newest version of C# is version 14 and the code reference is no longer part of DIKUArcade, so I assume this is no longer an issue to be concerned about.
